src/chain.h

```diff
@@ -600,7 +608,14 @@ class CDiskBlockIndex : public CBlockIndex
         // block header
         READWRITE(this->nVersion);
         READWRITE(hashPrev);
-        READWRITE(hashVeilData);
+        bool fCheckVeilDataHash = false;
+        if (nVersion >> 28 != 3) {
```

Comment:
Can you explain the logic here; may need more comments.  From what I can tell, current version would match this check (0x20000000), and so I presume new version is 0x30000000.  I would much prefer to see this the VERSIONBITS values and masks.  I also am curious to understand more about what "version bits" are compared to regular "pre-version bits" versions.

Reply:
@mimirmim I am referring to:
```
static const int32_t VERSIONBITS_LAST_OLD_BLOCK_VERSION = 4;
/** What bits to set in version for versionbits blocks */
-static const int32_t VERSIONBITS_TOP_BITS = 0x04000000UL;
+static const int32_t VERSIONBITS_TOP_BITS = 0x01000000UL;
+static const int32_t VERSIONBITS_OLD_POW_VERSION = 0x20000000UL;
+static const int32_t VERSIONBITS_NEW_POW_VERSION = 0x30000000UL;
/** What bitmask determines whether versionbits is in use */
-static const int32_t VERSIONBITS_TOP_MASK = 0xFC000000UL;
+static const int32_t VERSIONBITS_TOP_MASK = 0xFF000000UL;
/** Total bits available for versionbits */
-static const int32_t VERSIONBITS_NUM_BITS = 26;
+static const int32_t VERSIONBITS_NUM_BITS = 24;
```

So this code `if (nVersion >> 28 != 3)` appears to be built on the tribal knowledge that comes from versionbits.h.  So I'm raising one core review comment, and one possible issue:

- This logic should be using the defines presented in versionbits.h and not magic numbers.
- Next time nVersion changes, this section of code will then execute.  It is not limited to cover "all the nVersions that will have hashVeilData and not any other future versions".

Reply:
I will update this not to use magic number. 
Also, I will update this to check for version 2. The current version and write it when only it is version 2.

Reply:
I have updated the code to not use magic number. Also, I change the check to look for version less than 2. Which includes the genesis block which has a version of 1. All new version going forward will now be supported. 

It isn't often that you need to change the block verison though. So that is great news.